tuf/snapshot.py

```diff
@@ -0,0 +1,258 @@
+import abc
+import os
+import logging
+
+from securesystemslib import formats as sslib_formats
+
+from tuf import formats
+from tuf import roledb
+from tuf import repository_lib as repolib
+
+# See 'log.py' to learn how logging is handled in TUF.
+logger = logging.getLogger(__name__)
+
+# The extension of TUF metadata.
+METADATA_EXTENSION = '.json'
+
+# The metadata filenames of the top-level roles.
+ROOT_FILENAME = 'root' + METADATA_EXTENSION
+TARGETS_FILENAME = 'targets' + METADATA_EXTENSION
+SNAPSHOT_FILENAME = 'snapshot' + METADATA_EXTENSION
+TIMESTAMP_FILENAME = 'timestamp' + METADATA_EXTENSION
+
+class SnapshotInterface():
```

Comment:
Can we distinguish between:

- the repository state (i.e., all metadata about all current packages)
- the artifact that clients will download

I think that will help make the roles/names of various classes clearer.